File: Backend/utils/validation.js
const validateWaterQualityData = (data) => {
  const errors = [];
  if (!data.tanggal) errors.push('Tanggal harus diisi');
  if (!data.lokasi) errors.push('Lokasi harus diisi');
  if (data.suhu === undefined || data.suhu < 0 || data.suhu > 45) errors.push('Suhu tidak valid');
  if (data.pH === undefined || data.pH < 0 || data.pH > 14) errors.push('pH tidak valid');
  if (data.salinitas === undefined || data.salinitas < 0 || data.salinitas > 50) errors.push('Salinitas tidak valid');
  if (data.kekeruhan === undefined || data.kekeruhan < 0) errors.push('Kekeruhan tidak valid');
  return errors;
};

const validateLogin = (data) => {
  const errors = [];
  if (!data.username) errors.push('Username harus diisi');
  if (!data.password) errors.push('Password harus diisi');
  return errors;
};

const validateUser = (data) => {
  const errors = [];
  if (!data.username) errors.push('Username harus diisi');
  if (!data.email) errors.push('Email harus diisi');
  if (!data.password || data.password.length < 6) errors.push('Password minimal 6 karakter');
  return errors;
};

module.exports = { validateWaterQualityData, validateLogin, validateUser };

File: backend_python/utils/validation.py
def validate_water_quality_data(data):
    errors = []
    if not data.get("tanggal"):
        errors.append("Tanggal harus diisi")
    if not data.get("lokasi"):
        errors.append("Lokasi harus diisi")
    if data.get("suhu") is None or data["suhu"] < 0 or data["suhu"] > 45:
        errors.append("Suhu tidak valid")
    if data.get("pH") is None or data["pH"] < 0 or data["pH"] > 14:
        errors.append("pH tidak valid")
    if data.get("salinitas") is None or data["salinitas"] < 0 or data["salinitas"] > 50:
        errors.append("Salinitas tidak valid")
    if data.get("kekeruhan") is None or data["kekeruhan"] < 0:
        errors.append("Kekeruhan tidak valid")
    return errors


def validate_login(data):
    errors = []
    if not data.get("username"):
        errors.append("Username harus diisi")
    if not data.get("password"):
        errors.append("Password harus diisi")
    return errors


def validate_user(data):
    errors = []
    if not data.get("username"):
        errors.append("Username harus diisi")
    if not data.get("email"):
        errors.append("Email harus diisi")
    if not data.get("password") or len(str(data["password"])) < 6:
        errors.append("Password minimal 6 karakter")
    return errors
